02-peer-protocol.md

```diff
@@ -577,17 +577,17 @@ Closing happens in two stages:
         |       | <complete all pending HTLCs> |       |
         |   A   |                 ...          |   B   |
         |       |                              |       |
-        |       |--(3)-- closing_signed  F1--->|       |
-        |       |<-(4)-- closing_signed  F2----|       |
-        |       |              ...             |       |
-        |       |--(?)-- closing_signed  Fn--->|       |
-        |       |<-(?)-- closing_signed  Fn----|       |
+        |       |--(3a)- closing_complete Fee->|       |
+        |       |<-(3b)- closing_complete Fee--|       |
+        |       |<-(4a)- closing_sig ----------|       |
+        |       |--(4b)- closing_sig --------->|       |
```

Comment:
This doesn't seem like the typical use case where one side initiates the close and the other signs.  Some explanation somewhere is needed:

- Is the idea here that both sides can contribute to the closing fee if they want?
- Does B's `closing_complete` override A's?  And in that case, why does B send `closing_sig`?
- Did both A and B send `closing_complete` at the same time, so we have 2 flows happening at once?

Reply:
There is one flow happening in each direction, where the sender of `closing_complete` is paying the fee.
Let me know if the proposed diagram [here](https://github.com/lightning/bolts/pull/1096#issuecomment-1743283241) makes it easier to reason about.

Reply:
> There is one flow happening in each direction, where the sender of `closing_complete` is paying the fee. 

Do we really expect dual flows to happen like that in practice?  If the other party offers to pay for the closing transaction, why would we want to send `closing_complete` at all?

> Let me know if the proposed diagram [here](https://github.com/lightning/bolts/pull/1096#issuecomment-1743283241) makes it easier to reason about.

Your diagram helps to clarify corner cases but probably still needs some high-level explanation in the spec.

Reply:
> Do we really expect dual flows to happen like that in practice?

I think we do, because each side should independently decide what fee they'd like to pay and immediately send `closing_complete` (at least that's how I implemented it). Since the channel has already stopped relaying HTLCs at that point, both peers are likely eager to get their funds back asap to reinvest them into another channel.